keytree/src/lib.rs

```diff
@@ -50,6 +50,36 @@ impl Xprv {
             precompressed_pubkey: self.precompressed_pubkey,
         }
     }
+
+    /// Serializes this Xprv to a sequence of bytes.
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut buf = [0u8; 64];
+        buf[..32].copy_from_slice(&self.scalar.to_bytes());
+        buf[32..].copy_from_slice(&self.dk);
+        buf
+    }
+
+    /// Decodes an Xprv from a 64-byte array, and fails if the provided array is not
+    /// exactly 64 bytes.
+    pub fn from_bytes(bytes: &[u8]) -> Option<Self> {
+        if bytes.len() != 64 {
+            return None;
+        }
+
+        let pieces = bytes.split_at(32);
+        let mut scalar_bytes: [u8; 32] = Default::default();
+        scalar_bytes.copy_from_slice(&pieces.0[..]);
+        let scalar = Scalar::from_bits(scalar_bytes);
```

Comment:
`Scalar::from_bits` is a function for specific use of bit-twiddling-infested protocols originally spawned by djb (x25519 and EdDSA). I'd recommend using `Scalar::from_canonical_bytes` that checks that a scalar is canonical. Our decoding func is already failable, so it's not going to make API uglier to fail on non-canonical scalars.

```suggestion
        let scalar = match Scalar::from_canonical_bytes(scalar_bytes) { Some(x)=>x, None => return None };
```